lib/puppet/functions/hiera_vault.rb

```diff
@@ -10,12 +10,12 @@
   begin
     require 'json'
   rescue LoadError => e
-    raise Puppet::DataBind::LookupError, "Must install json gem to use hiera-vault"
+    raise Exception, "[hiera-vault] Must install json gem to use hiera-vault backend"
```

Comment:
use of `Puppet::DataBinding::LookupError` was fine, that is what similar lookup functions are using.

For other Exceptions, see this diagram http://rubylearning.com/satishtalim/ruby_exceptions.html. If you have no specific exception to raise, just use `raise "there was a problem..."` and that will raise a StandardError.  If you need to create your own exceptin types, derive them from `StandardError`. Never use `Exception` - as you can see the errors derived from it (except StandardError) are for specific runtime and very bad conditions.